### PR TITLE
EffectCommandEvent: SkriptEvent support

### PR DESCRIPTION
**Pull request by NotSoDelayed**
### Description
This PR adds support for Skript users to listen to EffectCommandEvent, and cleans up the involved classes.

Event values are:
- `event-sender` returns the sender (i.e. Player, Console)
- `[the] command` (from ExprCommand) returns the involved command

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/events/EvtEffectCommand.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		Executor(String toString) {
+			this.toString = toString;
+		}
+
+		@Override
+		public String toString() {
+			return toString;
+		}
+
+	}
```
[the remark this answers is not in the document]
Comment (NotSoDelayed) [reply src/main/java/ch/njol/skript/events/EvtEffectCommand.java:61]: I can see this implementation will definitely put into a good use!

[the remark this answers is not in the document]
Comment (sovdeeth) [reply src/main/java/ch/njol/skript/events/EvtEffectCommand.java:61]: I'd vastly prefer an Enum, a Kleenean doesn't make much contextual sense to me. That said, you could simplify the logic a bit by using parse marks and the ordinal of the enum, if you wanted.

File: src/main/java/ch/njol/skript/expressions/ExprCommand.java [hunk 2]
```diff
@@ -72,23 +74,25 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
...
-			ScriptCommandEvent event = (ScriptCommandEvent) e;
-			s = event.getCommandLabel() + " " + event.getArgsString();
+		if (event instanceof PlayerCommandPreprocessEvent) {
+			command = ((PlayerCommandPreprocessEvent) event).getMessage().substring(1).trim();
+		} else if (event instanceof ServerCommandEvent) {
+			command = ((ServerCommandEvent) event).getCommand().trim();
+		} else if (event instanceof ScriptCommandEvent) {
+			ScriptCommandEvent e = (ScriptCommandEvent) event;
+			command = e.getCommandLabel() + " " + e.getArgsString();
+		} else { // It's an EffectCommandEvent
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/expressions/ExprCommand.java:87]: best to make sure

File: src/main/java/ch/njol/skript/events/EvtEffectCommand.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+						"on effect command:",
+								"\tlog \"%sender%: %command%\" to file \"effectcommand.log\"")
+				.since("INSERT VERSION");
+	}
+
+	private enum Executor {
+
+		ANY(""),
+		CONSOLE("console "),
+		PLAYER("player ");
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/events/EvtEffectCommand.java:48]:
```suggestion
		CONSOLE("console"),
		PLAYER("player");
```
spacing should be handled by the caller.

**Review by sovdeeth — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/events/EvtEffectCommand.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+public class EvtEffectCommand extends SkriptEvent {
+
+	static {
+		Skript.registerEvent("Effect Command Event", EvtEffectCommand.class, EffectCommandEvent.class,
+				"[sender:(:player|console)] effect command")
+				.description("Called when a player or console executes a skript effect command.")
+				.examples(
+						"on effect command:",
+								"\tlog \"%sender%: %command%\" to file \"effectcommand.log\"")
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/events/EvtEffectCommand.java:40]:
```suggestion
							"\tlog \"%sender%: %command%\" to file \"effectcommands.log\"")
```

**Comment by NotSoDelayed** [comment]
Closing, in favour of #6587

**Pull request closed without merging**